interop-reactiveStreams/shared/src/main/scala/scalaz/zio/interop/reactiveStreams/StreamPublisher.scala

```diff
@@ -18,35 +18,19 @@ class StreamPublisher[R, E <: Throwable, A](
           demand  <- Queue.unbounded[Long]
           _       <- Task(subscriber.onSubscribe(createSubscription(subscriber, demand)))
           control = Stream.fromQueue(demand).flatMap(n => Stream.unfold(n)(n => if (n > 0) Some(((), n - 1)) else None))
-          fiber   <- stream.toQueue().use(takesToCallbacks(subscriber, _, control, demand)).fork
+          fiber <- stream
+                    .zip(control)
+                    .foreach { case (a, _) => Task(subscriber.onNext(a)) }
+                    .map(_ => subscriber.onComplete())
```

Comment:
Since this is a side-effecting operation that may throw, you'd want to formulate this as `.flatMap(_ => Task(subscriber.onComplete())`. Throwing in `map` would result in the fiber terminating with a defect, whereas throwing inside `Task` would result in the fiber failing normally.